pytential/symbolic/pde/scalar.py

```diff
@@ -190,53 +264,61 @@ def __init__(self, kernel: Kernel, loc_sign: int, *,
             else:
                 alpha = 0
 
-        super().__init__(kernel, use_l2_weighting)
+        super().__init__(kernel, use_l2_weighting=use_l2_weighting)
 
         self.kernel_arguments = kernel_arguments
         self.loc_sign = loc_sign
         self.alpha = alpha
 
-    def is_unique_only_up_to_constant(self):
+    def is_unique_only_up_to_constant(self) -> bool:
         # No ones matrix needed in Helmholtz case, cf. Hackbusch Lemma 8.5.3.
         from sumpy.kernel import LaplaceKernel
         return isinstance(self.kernel, LaplaceKernel) and self.loc_sign > 0
 
-    def representation(self, u,
-            map_potentials=None, qbx_forced_limit=None, **kwargs):
-        """
-        :param u: symbolic variable for the density.
-        :param map_potentials: a callable that can be used to apply
-            additional transformations on all the layer potentials in the
-            representation, e.g. to take a target derivative.
-        :param kwargs: additional keyword arguments passed on to the layer
-            potential constructor.
-        """
-        sqrt_w = self.get_sqrt_weight()
+    @override
+    def representation(self,
+                       u: ArithmeticExpression,
+                       *,
+                       map_potentials: PotentialMapper | None = None,
+                       qbx_forced_limit: QBXForcedLimit = None,
+                       source: DOFDescriptorLike = None,
+                       target: DOFDescriptorLike = None,
+                       **kwargs: Operand) -> ArithmeticExpression:
+        sqrt_w = self.get_sqrt_weight(source)
         inv_sqrt_w_u = sym.cse(u/sqrt_w)
 
         if map_potentials is None:
-            def map_potentials(x):
+            def default_map_potentials(x: ArithmeticExpression) -> ArithmeticExpression:
                 return x
 
-        kwargs["kernel_arguments"] = self.kernel_arguments
-        kwargs["qbx_forced_limit"] = qbx_forced_limit
-
-        return (
-                self.alpha * map_potentials(
-                    sym.S(self.kernel, inv_sqrt_w_u, **kwargs)
-                    )
-                - map_potentials(
-                    sym.D(self.kernel, inv_sqrt_w_u, **kwargs)
-                    )
-                )
-
-    def operator(self, u, **kwargs):
-        """
-        :param u: symbolic variable for the density.
-        :param kwargs: additional keyword arguments passed on to the layer
-            potential constructor.
-        """
-        sqrt_w = self.get_sqrt_weight()
+            map_potentials = default_map_potentials
+
+        kernel_arguments = self.kernel_arguments
+        if kwargs:
+            kernel_arguments = {**kernel_arguments, **kwargs}
```

Comment:
Before, `kwargs` and `kernel_arguments` would get passed down to `IntG`, which would raise a `ValueError` if some keys repeated. From a quick search, it doesn't seem like anyone is using those extra `kwargs` anyway, but is it ok to let it overwrite?

Reply:
Hm, actually `int_g_vec` already does exactly this since #242 (also my bad!). So I guess these things got overwritten for a while now..